assets/my-first-radanalytics-app/sparkpi-scala-scalatra.adoc

```diff
@@ -0,0 +1,366 @@
+
+[source,scala]
+----
+package io.radanalytics.examples.scalatra.sparkpi
+
+import org.eclipse.jetty.server.Server
+import org.eclipse.jetty.servlet.DefaultServlet
+import org.eclipse.jetty.webapp.WebAppContext
+import org.scalatra.servlet.ScalatraListener
+
+object Main {
+
+    def main( args : Array[ String ] ) : Unit = {
+        val port = 8080
+        val server = new Server( port )
+        val context = new WebAppContext()
+
+        context.setContextPath( "/" )
+        context.setResourceBase( "src/main/webapp" )
+        context.setInitParameter( ScalatraListener.LifeCycleKey, "io.radanalytics.examples.scalatra.sparkpi.ScalatraInit" ) // Override the Scalatra default for ScalatraBootstrap in default package
+        context.addEventListener( new ScalatraListener )
+
+        server.setHandler( context )
+        server.start()
+        server.join()
+    }
+
+}
+----
+
+== Build the Application and Verify Locally
+The directory structure should now look like this:
+
+----
+tutorial-sparkpi-scala-scalatra (root)
+    \_ build.sbt
+    \_ project
+        \_ Dependencies.scala
+        \_ build.properties
+        \_ plugins.sbt
+    \_ src
+        \_ main
+            \_ scala
+                \_ io
+                    \_ radanalytics
+                        \_ examples
+                            \_ scalatra
+                                \_ sparkpi
+                                    \_ Main.scala
+                                    \_ ScalatraInit.scala
+                                    \_ SparkPi.scala
+                                    \_ SparkPiServlet.scala
+        \_ test
+            \_ scala
+                \_ io
+                    \_ radanalytics
+                        \_ examples
+                            \_ scalatra
+                                \_ sparkpi
+                                    \_ SparkPiTest.scala
+----
+
+At this point it is possible to build and unit test the application locally. In the context of this tutorial, which is to deploy the application to Openshift, this step is optional. However, it can be helpful in validating that the code builds and passes all the tests before being deployed. If you have SBT installed, you can run the tests and generate the executable jar with the following command:
+
+[source,shell]
+----
+# if this is your first SBT build this could take quite some time.
+sbt clean test assembly
+----
+
+If this build succeeds, the deployable "fat jar" will be located in the following location:
+
+----
+tutorial-sparkpi-scala-scalatra (root)
+    \_ build.sbt
+    ...
+    \_ target
+        \_scala-2.11
+            \_ tutorial-sparkpi-scala-scalatra-assembly-0.0.1-SNAPSHOT.jar
+
+----
+
+This deployment can be tested loby running `java -jar target/scala-2.11/tutorial-sparkpi-scala-scalatra-assembly-0.0.1-SNAPSHOT.jar. The service should start, but will fail when servicing requests, unless you have a Spark cluster running locally.
```

Comment:
missing a closing quote on the "java -jar..." part

Reply:
@elmiko sorry got busy with something else. I'll take care of this tonight.

Reply:
no apologies necessary. i know the feeling all too well ;)